src/js/options.js

```diff
@@ -456,15 +467,24 @@ function removeWhitelistDomain(event) {
  */
 function getOrigins() {
   var origins = {};
-  var action_map = badger.storage.getBadgerStorageObject('action_map');
+  var action_map = OPTIONS_DATA.action_map;
+  let messageSends = [];
   for (var domain in action_map.getItemClones()) {
```

Comment:
This fails because sending the entire `action_map` objects loses its methods in serialization. Instead of sending all of `action_map`, move the logic of gathering origins to `webrequest.js`, and then send the origins all at once (instead of sending hundreds of individual messages/using Promises).

Reply:
Done